README.md

```diff
@@ -340,32 +353,292 @@ The above route will just return the string "Hello World!" in the web browser wh
-from flask import render_template
+```{.python file=src/py/hello-templated.py}
+from flask import Flask, render_template
 
-@app.route("/")
-def newtonraphson():
-    return render_template('newtonraphson.html')
+app = Flask(__name__)
+
+@app.route('/hello/<name>')
+def hello_name(name=None):
+    return render_template('hello.html', name=name)
+
+app.run()
 ```
 
 Where `name` is a variable which gets combined with template to render into a html page.
 
+The web application can be started with
+```{.awk #py-hello-templated}
+python src/py/hello-templated.py
+```
+
+In a web browser you can visit http://localhost:5000/hello/yourname to the web application.
+
+Let's make the web application for our Newton raphson algorithm.
+
+The first thing we want is the web page with the form, the template that renders the form looks like
+
+```{.html file=src/py/templates/form.html}
+<!doctype html>
+<form method="POST">
+  <label for="epsilon">Epsilon</label>
+  <input type="number" name="epsilon" value="0.001">
+  <label for="guess">Guess</label>
+  <input type="number" name="guess" value="-20">
+  <button type="submit">Submit</button>
+</form>
+```
+
+The home page will render the form like so
+
+```{.python #py-form}
+@app.route('/', methods=['GET'])
+def form():
+  return render_template('form.html')
+```
+
+The result will be displayed on a html page with the following template
+
+```{.html file=src/py/templates/result.html}
+<!doctype html>
+<p>With epsilon of {{ epsilon }} and a guess of {{ guess }} the found root is {{ root }}.</p>
+```
+
+The form will be submitted to the '/' path with the POST method. In the handler of this route we want to perform the calculation and return the result html page.
+
+```{.python #py-calculate}
+@app.route('/', methods=['POST'])
+def calculate():
+  epsilon = float(request.form['epsilon'])
+  guess = float(request.form['guess'])
+
+  from newtonraphsonpy import NewtonRaphson
+  finder = NewtonRaphson(epsilon)
+  root = finder.find(guess)
+
+  return render_template('result.html', epsilon=epsilon, guess=guess, root=root)
+```
+
+Putting it all together in
+
+```{.python file=src/py/webapp.py}
+from flask import Flask, render_template, request
+app = Flask(__name__)
+
+<<py-form>>
+
+<<py-calculate>>
+
+app.run()
+```
+
+And running it with
+
+```{.awk #py-webapp}
+PYTHONPATH=$PWD python src/py/webapp.py
+```
+
+To test we can visit http://localhost:5000 fill the form and press submit to get the result.
+
 When performing a long calculation (more than 30 seconds), the end-user requires feedback of the progress. In a normal request/response cycle, feedback is only returned in the response. To give feedback during the calculation, the computation must be offloaded to a task queue. In Python the most used task queue is [celery](http://www.celeryproject.org/). While the calculation is running on some worker it is possible to have a progress page which can check in the queue what the progress is of the calculation.
```

Comment:
Can we mention redis and celery in the beginning? Or in the introduction say something like: 'this will be explained in following sections' and give a reference.

Reply:
Inserted chapter header to more clearly separate that now we will be talking about long running tasks. Is that better?